x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/secrets/ConnectorSecretsIndexService.java

```diff
@@ -93,4 +95,15 @@ public void createSecret(PostConnectorSecretRequest request, ActionListener<Post
             listener.onFailure(e);
         }
     }
+
+    public void deleteSecret(String id, ActionListener<DeleteConnectorSecretResponse> listener) {
+        clientWithOrigin.prepareDelete(CONNECTOR_SECRETS_INDEX_NAME, id)
```

Comment:
is there any scenario in which we want to wrap this in `try/except` ? The API logic is responsible for handling it's own exceptions, any unhandled exception here can fail the node